src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php

```diff
@@ -347,6 +340,33 @@ public function handleVetSecondFactorCommand(VetSecondFactorCommand $command)
         $this->eventSourcedRepository->save($registrant);
     }
 
+    public function handleSelfVetSecondFactorCommand(SelfVetSecondFactorCommand $command)
+    {
+        /** @var IdentityApi $identity */
+        $identity = $this->eventSourcedRepository->load(new IdentityId($command->identityId));
+        $secondFactorIdentifier = SecondFactorIdentifierFactory::forType(
+            new SecondFactorType($command->secondFactorType),
+            $command->secondFactorId
+        );
+        $loa = $this->loaResolutionService->getLoa($command->authoringSecondFactorIdentifier);
```

Comment:
Domain logic should reside in aggregates and not in command handlers, please move this to the Identity aggregate.

Reply:
I misused the domain exception here. I don't view this as domain logic, but more like input validation/input dressing. The alternative would be to inject the loa resolution helper onto the identity aggregate, and that would introduce other undesirable issues.